apps/ensnode/src/handlers/Registry.ts

```diff
@@ -76,172 +77,179 @@ async function recursivelyRemoveEmptyDomainFromParentSubdomainCount(context: Con
   }
 }
 
-export async function handleTransfer({
-  context,
-  event,
-}: {
-  context: Context;
-  event: EventWithArgs<{ node: Hex; owner: Hex }>;
-}) {
-  const { node, owner } = event.args;
-
-  // Each domain must reference an account of its owner,
-  // so we ensure the account exists before inserting the domain
-  await upsertAccount(context, owner);
-
-  // ensure domain & update owner
-  await context.db
-    .insert(schema.domain)
-    .values([{ id: node, ownerId: owner, createdAt: event.block.timestamp }])
-    .onConflictDoUpdate({ ownerId: owner });
-
-  // garbage collect newly 'empty' domain iff necessary
-  if (owner === zeroAddress) {
-    await recursivelyRemoveEmptyDomainFromParentSubdomainCount(context, node);
-  }
-
-  // log DomainEvent
-  await context.db.insert(schema.transfer).values({
-    ...sharedEventValues(event),
-    domainId: node,
-    ownerId: owner,
-  });
-}
-
-export const handleNewOwner =
-  (isMigrated: boolean) =>
-  async ({
-    context,
-    event,
-  }: {
-    context: Context;
-    event: EventWithArgs<{ node: Node; label: Labelhash; owner: Hex }>;
-  }) => {
-    const { label, node, owner } = event.args;
-
-    const subnode = makeSubnodeNamehash(node, label);
-
-    // Each domain must reference an account of its owner,
-    // so we ensure the account exists before inserting the domain
-    await upsertAccount(context, owner);
-
-    // note that we set isMigrated so that if this domain is being interacted with on the new registry, its migration status is set here
-    let domain = await context.db.find(schema.domain, { id: subnode });
-    if (domain) {
-      // if the domain already exists, this is just an update of the owner record (& isMigrated)
-      await context.db.update(schema.domain, { id: domain.id }).set({ ownerId: owner, isMigrated });
-    } else {
-      // otherwise create the domain
-      domain = await context.db.insert(schema.domain).values({
-        id: subnode,
+export const makeRegistryHandlers = (ownedName: OwnedName) => {
```

Comment:
I had to wrap multiple exports into a factory function that would allow passing extra context to those previously-exported functions. Haven't changed logic of any of these functions.